suite/src/modules/opentitan_hmac.rs

```diff
@@ -0,0 +1,167 @@
+#![allow(dead_code)]
+
+use crate::modules::{Module, SHA256Module};
+use bitflags::bitflags;
+
+bitflags! {
+    /// Abstract representation of the config registers flags.
+    struct HmacCFG: u32 {
+        const HMAC_ENABLED = 1 << 0;
+        const SHA_ENABLED = 1 << 1;
+        /// If set the input is interpreted in little endian, otherwise big endian
+        const ENDIAN_SWAPPED = 1 << 2;
+        /// If set the output is in big endian, otherwise little endian
+        const DIGEST_ENDIAN_SWAPPED = 1 << 3;
+    }
+
+    /// Abstract representation of the command registers flags.
+    struct HmacCMD: u32 {
+        const HASH_START = 1 << 0;
+        const HASH_PROCESS = 1 << 1;
+    }
+
+    /// Abstract representation of the status registers flags.
+    struct HmacSTATUS: u32 {
+        const FIFO_EMPTY = 1 << 0;
+        const FIFO_FULL = 1 << 1;
+    }
+
+    /// Abstract representation of the interrupt state registers flags.
+    struct HmacINTRSTATE: u32 {
+        const HMAC_DONE = 1 << 0;
+        const FIFO_EMPTY = 1 << 1;
+        const HMAC_ERR = 1 << 2;
+    }
+}
+
+/// Offset of the interrupt state register
```

Comment:
I am assuming you have double checked the register map?

Reply:
Yes they should all match the information provided by the register table in the [documentation].(https://docs.opentitan.org/hw/ip/hmac/doc/#register-table)